### PR TITLE
fix: Fixed Web Chat tab switching behavior

### LINKED ISSUE #8365: Web Chat log opens when you click on a card, making it difficult to use Watch
## Describe the bug

When you click on a message, the Web Chat log was designed to show you the log item that corresponds to that message. This can cause problems if you are watching a conversation with cards: if your Watch pivot is selected, when you click on a button or element in a card, the Web Chat pivot becomes selected so the Watch pivot is hidden.

## Version

2.0.0-nightly.258070.287a4dc

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [x] macOS
- [ ] Windows
- [ ] Ubuntu

## To Reproduce

1. Open a bot with cards, like Enterprise Assistant.
2. Open the Watch pane and add some variables.
3. Start your bot and interact with the cards.
4. You will see that when you click on a card, that card become in focus and the Web Chat pivot becomes active (hiding the Watch pane), and the log related to the card is selected.

## Expected behavior

1 - When clicking on a button or input in a card, do not select the message.
2 - When clicking on a message, select the corresponding Web Chat log item, but do not make the Web Chat pivot the selected pivot.

## Screenshots

https://user-images.githubusercontent.com/39317542/125357469-88e17b00-e31c-11eb-8890-f047768164bb.mov

### PR DESCRIPTION
**Pull request by tonyanziano**
## Description

Clicking / focusing a message in the Web Chat panel, currently opens the debug panel if it is closed, and switches from whatever tab you are on to the Web Chat tab.

This PR stops the message selection from switching the active tab, unless it is your first time opening the debug panel.

## Task Item

fixes #8365 (P0)

## Screenshots

WatchTabP0Fix

### CODE REVIEW

File: Composer/packages/client/src/components/WebChat/hooks/useTranscriptFocusListener.ts [hunk 1]
```diff
@@ -33,12 +34,15 @@ export const useTranscriptFocusListener = () => {
         }
       });
       if (trafficItem && currentProjectId) {
-        setDebugPanelActiveTab(WebChatInspectorTabKey);
+        // only switch to the Web Chat tab if no tab is active
```
Comment (hatpick) [inline Composer/packages/client/src/components/WebChat/hooks/useTranscriptFocusListener.ts:37]: comment is about the logic above this line: if activity (first arg) doesn't have a value, maybe return right away?

Comment (tonyanziano) [reply Composer/packages/client/src/components/WebChat/hooks/useTranscriptFocusListener.ts:37]: Are you talking about just adding:

```javascript
useCallback(
  ({ activity }: { activity: Activity }) => {
    if (!activity) {
      return;
    }
```

Comment (hatpick) [reply Composer/packages/client/src/components/WebChat/hooks/useTranscriptFocusListener.ts:37]: yep, what do you think?

Comment (tonyanziano) [reply Composer/packages/client/src/components/WebChat/hooks/useTranscriptFocusListener.ts:37]: I don't think the [hook ever gets called with an `undefined` value](https://github.com/microsoft/BotFramework-WebChat/blob/3fc2853cc53bd4cf49950862ede25af7f923e268/packages/component/src/hooks/useObserveTranscriptFocus.ts#L7), but it doesn't hurt

Comment (hatpick) [reply Composer/packages/client/src/components/WebChat/hooks/useTranscriptFocusListener.ts:37]: try adding a hero card with actions and click those action buttons, activity seems ```undefined``` at that point

Comment (tonyanziano) [reply Composer/packages/client/src/components/WebChat/hooks/useTranscriptFocusListener.ts:37]: let me give that a shot

Comment (tonyanziano) [reply Composer/packages/client/src/components/WebChat/hooks/useTranscriptFocusListener.ts:37]: You're right! I'll make the change

**Review by hatpick — APPROVED** [review]

**Pull request merged**